homeassistant/components/sensor/envirophat.py

```diff
@@ -53,7 +53,13 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Set up the Sense HAT sensor platform."""
-    data = EnvirophatData(config.get(CONF_USE_LEDS))
+    try:
```

Comment:
@pvizeli 
I'm guessing I'm a bit confused of the aim of this part. Catching the `ImportError` would be appropriate in the case when the library is not actually installed, and that needs to be found.

But the issue trying to be addressed is whether or not the actual Enviro pHAT board is attached. (as in the original comment by @fabaff , if I understand it correctly https://github.com/home-assistant/home-assistant/pull/7427#discussion_r114703904 )

In the cases of:

1) Library not installed - `ImportError` thrown, handled as you mentioned
2) Library installed but no Enviro pHAT attached - that will throw an OSError down the stack like this:

```
2017-05-09 20:52:03 ERROR (MainThread) [homeassistant.components.sensor] Error while setting up platform envirophat
Traceback (most recent call last):
  File "/mnt/nfsserver/home-assistant/homeassistant/helpers/entity_component.py", line 155, in _async_setup_platform
    entity_platform.schedule_add_entities, discovery_info
  File "/usr/lib/python3.4/asyncio/futures.py", line 388, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.4/asyncio/tasks.py", line 286, in _wakeup
    value = future.result()
  File "/usr/lib/python3.4/asyncio/futures.py", line 277, in result
    raise self._exception
  File "/usr/lib/python3.4/concurrent/futures/thread.py", line 54, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/mnt/nfsserver/home-assistant/homeassistant/components/sensor/envirophat.py", line 57, in setup_platform
    import envirophat
  File "/mnt/nfsserver/home-assistant/venv/lib/python3.4/site-packages/envirophat/__init__.py", line 12, in <module>
    light = tcs3472(bus)
  File "/mnt/nfsserver/home-assistant/venv/lib/python3.4/site-packages/envirophat/tcs3472.py", line 37, in __init__
    self.i2c_bus.write_byte_data(ADDR, REG_ENABLE, REG_ENABLE_RGBC | REG_ENABLE_POWER)
  File "/mnt/nfsserver/home-assistant/venv/lib/python3.4/site-packages/smbus/util.py", line 59, in validator
    return fn(*args, **kwdefaults)
  File "/mnt/nfsserver/home-assistant/venv/lib/python3.4/site-packages/smbus/smbus.py", line 145, in write_byte_data
    raise IOError(ffi.errno)
OSError: 121
```
and this 2nd message is what this code is trying to make understandable. In the current form it handles both cases (letting ImportError through as it is, and catching hardware communication error for clarity)

If after this, still the best way forward is just having a bare `import envirophat` with nor error handling nor logging of what caused the error, then I'll update to that. Just want to make sure that we are on the same page :)